### PR TITLE
Update changefeed docs for protected timestamps

### PR DESCRIPTION
**Pull request by kathancox**
Fixes DOC-9241

This PR updates the changefeed docs to include some updates to how (by default) changefeeds interact with PTS.

- New default cluster setting section
- Deprecation of `protect_data_from_gc_on_pause`
- Add the upgrade to v23.2 breaking change to the release notes.

Most new content rendered: [Protect Changefeed Data](https://deploy-preview-18087--cockroachdb-docs.netlify.app/docs/v23.2/protect-changefeed-data.html)

Updates to docs include moving the previous Garbage Collection and changefeeds section to its own page since this is a critical part of changefeed usage that customers need to understand.

### CODE REVIEW

File: src/current/v23.2/protect-changefeed-data.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+To prevent an accumulation of protected changes that could impact performance, consider the following:
+
+- [`changefeed.protect_timestamp.max_age`](#changefeed-protect_timestamp-max_age)
+- [`gc_protect_expires_after`](#gc_protect_expires_after)
+
+### `changefeed.protect_timestamp.max_age`
+
+{% include_cached new-in.html version="v23.2" %} By default, the `changefeed.protect_timestamp.max_age` [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}) sets the maximum time that changefeeds making no forward progress will hold protected timestamp records. Once the `changefeed.protect_timestamp.max_age` duration is reached, the changefeed will fail with a permanent error. As a result, it is **critical to monitor for changefeed failures** because changefeeds will eventually fail with an unrecoverable error if they cannot progress before the duration is reached.
+
+This cluster setting is enabled by default to 4 days.
```
Comment (miretskiy) [inline src/current/v23.2/protect-changefeed-data.md:28]: You can also mention (though this is in the setting doc string) -- that this functionality may be disabled by setting max_age to 0 (though, we recommend against that).

File: src/current/v23.2/protect-changefeed-data.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SET CLUSTER SETTING changefeed.protect_timestamp.max_age = '120h';
+~~~
+
+{{site.data.alerts.callout_info}}
+`changefeed.protect_timestamp.max_age` applies only to **newly created changefeeds in v23.2**.
+
+If you are [upgrading to v23.2]({% link {{ page.version.version }}/upgrade-cockroach-version.md %}), it is necessary to set `protect_data_from_gc_on_pause` on any existing changefeeds to ensure that it does not enter a situation of infinite retries, which could prevent garbage collection. You can use the [`ALTER CHANGEFEED`]({% link {{ page.version.version }}/alter-changefeed.md %}) statement to add `protect_data_from_gc_on_pause` to existing changefeeds.
```
Comment (miretskiy) [inline src/current/v23.2/protect-changefeed-data.md:40]: it's not strictly necessary;
We recommend that you consider setting this option (in order to opt into this functionality).

File: src/current/v23.2/protect-changefeed-data.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+The [`gc_protect_expires_after`]({% link {{ page.version.version }}/create-changefeed.md %}#gc-protect-expire) option automatically expires the protected timestamp records that are older than the defined duration and cancels a changefeed job.
+
+For example:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE CHANGEFEED FOR TABLE db.table INTO 'external://sink' WITH on_error='pause', gc_protect_expires_after='24h';
+~~~
+
+If this changefeed runs into a retryable error, protected timestamps will protect changes for up to 24 hours. After this point, if the changefeed has not resumed, the protected timestamp records will expire and the changefeed job will be canceled to prevent accumulation of garbage.
```
Comment (miretskiy) [inline src/current/v23.2/protect-changefeed-data.md:54]: "has not resumed" probably needs to be changed (changefeed does not need to be resumed -- it's probably running, and is retry once in a while to try to deliver a message)
Something like:
... If the changefeed has not made any progress in the past 24 hours, the protected ....

**Review by miretskiy — APPROVED** [review]

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md [hunk 4]
```diff
@@ -10,6 +10,7 @@ Release Date: September 26, 2023
...
 - When customizing the [SQL shell's interactive prompt]({% link v23.2/cockroach-sql.md %}), the special sequence `%M` now expands to the full host name instead of the combination of host name and port number. To include the port number explicitly, use `%>`. The special sequence `%m` now expands to the host name up to the first period. [#105137][#105137]
 - The [`cockroach debug zip`]({% link v23.2/cockroach-debug-zip.md %}) command stores data retrieved from SQL tables in the remote cluster using the TSV format by default. [#107474][#107474]
+- The [`changefeed.protect_timestamp.max_age` cluster setting]({% link v23.2/protect-changefeed-data.md %}) will only apply to newly created changefeeds in v23.2. For existing changefeeds, you must ensure that the `protect_data_from_gc_on_pause` option is set so that changefeeds do not experience infinite retries and accumulate protected change data. You can use the [`ALTER CHANGEFEED`]({% link v23.2/alter-changefeed.md %}) statement to add `protect_data_from_gc_on_pause` to existing changefeeds. [#103539][#103539]
```
Comment (florence-crl) [inline src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md:13]: Would you be able to add a link on the first occurrence of `protect_data_from_gc_on_pause` to https://www.cockroachlabs.com/docs/v23.1/create-changefeed#protect-pause?

File: src/current/v23.2/protect-changefeed-data.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+## Prevent garbage accumulation
+
+To prevent an accumulation of protected changes that could impact performance, consider defining an expiration duration:
+
+- [`changefeed.protect_timestamp.max_age`](#changefeed-protect_timestamp-max_age): a [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}) to define a protected timestamp expiration for all changefeeds on a cluster.
+- [`gc_protect_expires_after`](#gc_protect_expires_after): a [changefeed option]({% link {{ page.version.version }}/create-changefeed.md %}#options) to define a protected timestamp expiration for a changefeed.
+
+In general, a few hours to a few days are appropriate values for these settings. A lower protected timestamp expiration should not have adverse effects on your changefeed as long as the changefeed is running. However, if the changefeed pauses, you will need to [resume]({% link {{ page.version.version }}/resume-job.md %}) it before the defined expiration time. The value of this setting should reflect how much time the changefeed may remain paused before it is canceled.
```
Comment (florence-crl) [inline src/current/v23.2/protect-changefeed-data.md:24]: In "The value of this setting", what is "this" referring to?

File: src/current/v23.2/protect-changefeed-data.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SET CLUSTER SETTING changefeed.protect_timestamp.max_age = '120h';
+~~~
+
+{{site.data.alerts.callout_info}}
+`changefeed.protect_timestamp.max_age` applies only to **newly created changefeeds in v23.2**.
+
+If you are [upgrading to v23.2]({% link {{ page.version.version }}/upgrade-cockroach-version.md %}), we recommend setting `protect_data_from_gc_on_pause` on any existing changefeeds to ensure that it does not enter a situation of infinite retries, which could prevent garbage collection. You can use the [`ALTER CHANGEFEED`]({% link {{ page.version.version }}/alter-changefeed.md %}) statement to add `protect_data_from_gc_on_pause` to existing changefeeds.
```
Comment (florence-crl) [inline src/current/v23.2/protect-changefeed-data.md:42]: Would you be able to add a link on the first occurrence of `protect_data_from_gc_on_pause` to https://www.cockroachlabs.com/docs/v23.1/create-changefeed#protect-pause?

**Review by florence-crl — APPROVED** [review]
LGTM, minor non-blocking suggestions

File: src/current/v23.2/protect-changefeed-data.md (hunk 6 above)
Comment (kathancox) [reply src/current/v23.2/protect-changefeed-data.md]: Added!

File: src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md (hunk 4 above)
Comment (kathancox) [reply src/current/_includes/releases/v23.2/v23.2.0-alpha.1.md]: Added!

File: src/current/v23.2/protect-changefeed-data.md (hunk 3 above)
Comment (kathancox) [reply src/current/v23.2/protect-changefeed-data.md]: Done!

File: src/current/v23.2/protect-changefeed-data.md (hunk 2 above)
Comment (kathancox) [reply src/current/v23.2/protect-changefeed-data.md]: Changed

File: src/current/v23.2/protect-changefeed-data.md (hunk 1 above)
Comment (kathancox) [reply src/current/v23.2/protect-changefeed-data.md]: Added.

**Comment by kathancox** [comment]
TFTRs!

**Pull request merged**